### PR TITLE
Feat/home

### PR DESCRIPTION
**Pull request by sonishreyas**
# Change logs

- Added home page UI
- Added toast
- Added create post modal
- Added text and emoji
#Preview
https://deploy-preview-2--ripple-social.netlify.app/

### CODE REVIEW

File: src/components/Home/NewPostModal.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	const handleDismissPostModal = () => {
+		setShowPostModal(false);
+		setShowModal(false);
+	};
+
+	const handleEmojiContainer = () =>
+		showEmojiContainer
+			? setShowEmojiContainer(false)
+			: setShowEmojiContainer(true);
+	// console.log(postState);
```
Comment (vidhi-thakur) [inline src/components/Home/NewPostModal.jsx:37]: console log can be removed here

File: src/components/Toast/ToastPortal.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			autoClose={1500}
+			hideProgressBar={false}
+			newestOnTop={false}
+			closeOnClick
+			rtl={false}
+			pauseOnFocusLoss
+			draggable
+			pauseOnHover
+		/>,
+		document.getElementById("toast-container")
```
Comment (vidhi-thakur) [inline src/components/Toast/ToastPortal.jsx:17]: I am not sure if this should be used here. Please confirm this with someone.

File: src/context/post-context.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+};
+const PostContext = createContext();
+
+const PostProvider = ({ children }) => {
+	const [postState, postDispatch] = useReducer(
+		postReducer,
+		defaultPostInitialState
+	);
+	const [showPostModal, setShowPostModal] = useState(false);
+	// console.log(postState);
```
Comment (vidhi-thakur) [inline src/context/post-context.js:19]: please remove

File: src/css/posts.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+.post-icons {
+  font-size: 1.8rem;
+}
+textarea {
+  border: none;
+  overflow: auto;
+  outline: none;
+
+  -webkit-box-shadow: none;
+  -moz-box-shadow: none;
```
Comment (vidhi-thakur) [inline src/css/posts.css:20]: nice 👍

**Review by vidhi-thakur — APPROVED** [review]
Code looks good, however, please confirm if we should be using `getElementById` code in React JS. Other than that, like the use of barrel exports, semantic tags and css variables

File: public/index.html [hunk 5]
```diff
@@ -22,7 +22,7 @@
     -->
   <meta name="author" content="Shreyas c. Soni" />
   <link rel="stylesheet" href="https://ripple-ui.netlify.app/css/components.css" />
-  <link rel="stylesheet" href="../src/css/index.css">
+  <link rel="stylesheet" href="../src/css/app.css">
```
Comment (ravikumar0403) [inline public/index.html:25]: files in src directory get's minified and bundled at build time and those files are added to a build directory, the build directory is the only thing that is deployed, so any files that you import from external folders will not be available unless it is served over the internet. Here your app.css is not being imported in deployed version. 
import it in your `App.js`

File: src/components/Toast/ToastPortal.jsx (hunk 2 above)
Comment (ravikumar0403) [reply src/components/Toast/ToastPortal.jsx:17]: This is correct, that's how you use react portals.

File: src/custom-hooks/useToast.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+				position: "top-right",
+				autoClose: 1500,
+				hideProgressBar: false,
+				closeOnClick: true,
+				pauseOnHover: true,
+				draggable: true,
+				progress: undefined,
+			});
+		};
+		notify();
```
Comment (ravikumar0403) [inline src/custom-hooks/useToast.jsx:20]: As these are the default styles of your toast, you can pass it as a prop to the ToastContainer component itself. imo, this abstraction is redundant and also adds a limitation of using toast only inside a component or custom hook.

File: src/reducers/modal-reducer.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const modalReducer = (state, { type, payload }) => {
+	switch (type) {
+		case "SET_MODAL":
+			return {
+				...state,
+				message: payload.message,
+				handleConfirm: payload.handleConfirm,
+				handleDismiss: payload.handleDismiss,
+				confirmChoice: payload.confirmChoice,
+				dismissChoice: payload.dismissChoice,
```
Comment (ravikumar0403) [inline src/reducers/modal-reducer.js:10]: The key in the payload is the same as the key in the state, so you can just destructed payload to achieve the same result

**Review by ravikumar0403 — CHANGES_REQUESTED** [review]
app.css is not applied to your deployed site, I've mentioned the reason in the comments.
Login functionality is not working. Check these out

**Pull request merged**